test/custom-parser.test.js

```diff
@@ -8,6 +8,27 @@ const sget = require('simple-get').concat
 const Fastify = require('..')
 
 const jsonParser = require('fast-json-body')
+
+function plainTextParser (request, callback) {
+  var body = ''
+  request.setEncoding('utf8')
+  request.on('error', onError)
+  request.on('data', onData)
+  request.on('end', onEnd)
+  function onError (err) {
+    callback(err, null)
+  }
+  function onData (chunk) {
+    body += chunk
+  }
+  function onEnd () {
+    setImmediate(parse, body)
```

Comment:
you can just do `callback(null, body)` no need to use `parse()` or `setImmediate()`

Reply:
OK, with code changed to...

```
function plainTextParser (request, callback) {
  var body = ''
  request.setEncoding('utf8')
  request.on('error', onError)
  request.on('data', onData)
  function onError (err) {
    callback(err, null)
  }
  function onData (chunk) {
    body += chunk
  }
  callback(null, body)
}
```

...a unit test fails;
```
test/custom-parser.test.js ........................ 130/131
  Can override the default plain text parser
  not ok should be equal
    +++ found
    --- wanted
    -hello world
    compare: ===
    at:
      line: 584
      column: 9
      file: test/custom-parser.test.js
      function: sget
    stack: |
      sget (test/custom-parser.test.js:584:9)
      concat (node_modules/simple-get/index.js:89:7)
      IncomingMessage.<anonymous> (node_modules/simple-concat/index.js:7:13)
    source: |
      t.strictEqual(body.toString(), 'hello world')

```

Reply:
you need to maintain onEnd()

Reply:
cheers @mcollina 
done. change pushed.

Reply:
@lfurzewaddock Flaky CI. Fixed now.

Reply:
:-)

Reply:
done.